lib/charts/renderers/waterfall_chart_renderer.dart

```diff
@@ -0,0 +1,206 @@
+      ..attr('width', dimensionScale.rangeBand - theme.defaultStrokeWidth)
+      ..attrWithCallback('y', (d, i, c) {
+      var tempY;
+      if (i <= ic && i > 0) {
+        tempY = prevY[order];
+        order++;
+      } else {
+        tempY = rect.height;
+      }
+      ic = i;
+      return tempY;
+    })
+      ..attr('height', 0)
+      ..on('click', (d, i, e) => _event(mouseClickController, d, i, e))
+      ..on('mouseover', (d, i, e) => _event(mouseOverController, d, i, e))
+      ..on('mouseout', (d, i, e) => _event(mouseOutController, d, i, e));
+
+    bar.transition()
+      ..styleWithCallback('fill', (d, i, c) => colorForKey(_reverseIdx(i)))
+      ..attr('width', dimensionScale.rangeBand - theme.defaultStrokeWidth)
+      ..duration(theme.transitionDuration);
+
+    var y = 0,
+    length = bar.length,
+    // Keeps track of heights of previously graphed bars. If all bars before
+    // current one have 0 height, the current bar doesn't need offset.
+    prevAllZeroHeight = true,
+    // Keeps track of the offset already exist in the previous bar, when the
+    // computed bar height is less than (theme.defaultSeparatorWidth +
+    // theme.defaultStrokeWidth), this height is already discounted, so the
+    // next bar's offset in height can be this much less than normal.
+    prevOffset = 0;
+
+    bar.transition()
+      ..attrWithCallback('y', (d, i, c) {
+      if (i == 0) y = measureScale.apply(0).round();
+      return (y -= (rect.height - measureScale.apply(d).round()));
+    })
+      ..attrWithCallback('height', (d, i, c) {
+      var ht = rect.height - measureScale.apply(d).round();
+      if (i != 0) {
+        // If previous bars has 0 height, don't offset for spacing
+        // If any of the previous bar has non 0 height, do the offset.
+        ht -= prevAllZeroHeight ? 1 :
+        (theme.defaultSeparatorWidth + theme.defaultStrokeWidth);
+        ht += prevOffset;
+      } else {
+        // When rendering next group of bars, reset prevZeroHeight.
+        prevOffset = 0;
+        prevAllZeroHeight = true;
+        ht -= 1;
+        // -1 so bar does not overlap x axis.
+      }
+      if (ht <= 0) {
+        prevOffset = prevAllZeroHeight ? 0 :
+        (theme.defaultSeparatorWidth + theme.defaultStrokeWidth) + ht;
+        ht = 0;
+      }
+      prevAllZeroHeight = (ht == 0) && prevAllZeroHeight;
+      return ht;
+    })
+      ..duration(theme.transitionDuration)
+      ..delay(50);
+
+    if (theme.defaultStrokeWidth > 0) {
+      enter.attr('stroke-width', '${theme.defaultStrokeWidth}px');
+      enter.styleWithCallback('stroke', (d, i, c) =>
+          colorForKey(_reverseIdx(i)));
+      bar.transition()
+        ..styleWithCallback('stroke', (d, i, c) => colorForKey(_reverseIdx(i)));
+    }
+
+    bar.exit.remove();
+  }
+
+  @override
+  double get bandInnerPadding =>
+      area.theme.dimensionAxisTheme.axisBandInnerPadding;
+
+  @override
+  Extent get extent {
```

Comment:
The max extent of the measure should probably be the highest point in the chart instead of the highest sum in a row (one stack).  In the case of a row's data needs to be subtracted (or negative) and also being the highest sum, it would return the wrong extent.

Does current implementation of waterfall renderer handle negative incremental value?